File: app/src/main/java/com/example/mike/mp3player/client/views/fragments/AsyncFragment.java
package com.example.mike.mp3player.client.views.fragments;

import android.os.Handler;
import android.os.Looper;

import androidx.fragment.app.Fragment;

import javax.inject.Inject;
import javax.inject.Named;

public abstract class AsyncFragment extends Fragment {

    Handler mainUpdater;
    Handler worker;

    @Inject
    public void setMainUpdater(@Named("main") Handler mainUpdater, @Named("worker") Handler worker) {
        this.mainUpdater = mainUpdater;
        this.worker = worker;
    }

    public AsyncFragment() {
        this.mainUpdater = new Handler(Looper.getMainLooper());
    }
}

File: app/src/main/java/com/github/goldy1992/mp3player/client/views/fragments/AsyncFragment.kt
package com.github.goldy1992.mp3player.client.views.fragments

import android.os.Handler
import android.os.Looper
import androidx.fragment.app.Fragment
import javax.inject.Inject
import javax.inject.Named

abstract class AsyncFragment : Fragment() {
    @JvmField
    var mainUpdater: Handler
    @JvmField
    var worker: Handler? = null
    @Inject
    fun setMainUpdater(@Named("main") mainUpdater: Handler, @Named("worker") worker: Handler?) {
        this.mainUpdater = mainUpdater
        this.worker = worker
    }

    init {
        mainUpdater = Handler(Looper.getMainLooper())
    }
}